erpnext/stock/doctype/delivery_note/delivery_note.py

```diff
@@ -898,7 +898,27 @@ def update_details(source_doc, target_doc, source_parent):
 		set_missing_values,
 	)
 
-	return doclist
+	# map Purchase Receipt with Purchase Order
+	for source_item in source_doc.items:
+		purchase_order_reference = frappe.get_value(
+			"Sales Order",
+			source_item.against_sales_order,
+			["inter_company_order_reference"]
+		)
+		purchase_order_item_details = frappe.get_all(
+			"Purchase Order Item",
+			filters={"parent": purchase_order_reference},
```

Comment:
This is a small optimization, but check if `purchase_order_reference` exists, otherwise it'll return all POs, I think